src/sinks/elasticsearch.rs

```diff
@@ -306,47 +306,26 @@ impl RetryLogic for ElasticSearchRetryLogic {
             )),
             _ if status.is_client_error() => {
                 let body = String::from_utf8_lossy(response.body());
-                warn!(
-                    message = "Client error",
-                    body = %body,
-                    rate_limit_secs = 30
-                );
-                RetryAction::DontRetry("client error".into())
+                RetryAction::DontRetry(format!("client-side error, {}: {}", status, body))
             }
             _ if status.is_success() => {
                 let body = String::from_utf8_lossy(response.body());
-                match body.find("\"errors\":true") {
-                    Some(_) => match serde_json::from_str::<ESResultResponse>(&body) {
-                        Err(json_error) => {
-                            warn!(
-                                message = "Elasticsearch unparsable error response",
-                                %json_error,
-                                rate_limit_secs = 30
-                            );
-                            RetryAction::DontRetry(
-                                "Some messages failed, and invalid response from Elasticsearch"
-                                    .into(),
-                            )
-                        }
-                        Ok(esrr) => {
-                            match esrr.items.into_iter().find_map(|item| item.index.error) {
-                                Some(error) => warn!(
-                                    message = "ElasticSearch error response",
-                                    err_type = %error.err_type,
-                                    reason = %error.reason,
-                                    rate_limit_secs = 30
-                                ),
-                                _ => warn!(
-                                    message = "Unusual ElasticSearch error response",
-                                    %body,
-                                    rate_limit_secs = 30
-                                ),
-                            };
-                            RetryAction::DontRetry("some messages failed".into())
+
+                if !body.contains("\"errors\":true") {
+                    return RetryAction::Successful;
+                }
+
+                let reason = match serde_json::from_str::<ESResultResponse>(&body) {
+                    Err(json_error) => format!("could not parse response, error: {}", json_error),
```

Comment:
I think this loses the "some messages failed" context. In particular, the error is not that we could not parse the response, the error is that ES told us it wouldn't accept all the events, and we couldn't parse the message they gave us to describe what went wrong.

Reply:
added "some messages failed".